### PR TITLE
Trigger blur event when window becomes inactive

### PR DESCRIPTION
**Pull request by davepagurek**
This addresses https://github.com/rsms/fb-mac-messenger/issues/133. Manually dispatching an event makes Messenger continue to send notifications as expected.

### CODE REVIEW

**Comment by rsms** [comment]
Have you verified that a "focus" event is always automatically emitted when window gains focus again? The blur & focus events obviously need to be reliable and balanced.

File: Messenger/AppDelegate.mm [hunk 1]
```diff
@@ -698,6 +698,11 @@ - (void)windowDidBecomeKey:(NSNotification*)notification {
   [self evaluateJavaScript:@"try { (typeof MacMessenger != 'undefined') && MacMessenger.focusComposer(); } catch(_) {}"];
 }
 
+- (void)windowDidResignKey:(NSNotification *)notification {
+  // Make sure a blur event is triggered when the app is on another workspace
```
Comment (rsms) [inline Messenger/AppDelegate.mm]: This is a misleading description — windowDidResignKey: is called every time the window is not the current user input target, like when switching to a different app in the same space or opening a different window in the same app (like the "about" window.)

If what you're looking to do is to emit a blur event when the window is on a different space (than the user's focus), you might want to look into leveraging different notifications and delegate methods.

Comment (davepagurek) [inline Messenger/AppDelegate.mm]: Right, this is called every time it goes into the background, the comment should clarify that the manual triggering is only needed for the case when it's in a different workspace. My intention was to call it every time it loses focus, just a bad comment

**Comment by davepagurek** [comment]
I haven't, I'll look into that and make it dispatch an event if needed. Agreed that it should be more reliable than this. Rather than detecting when the app is in another workspace, I think I'll try seeing if a blur event is dispatched after we know the app lost focus, and if not, trigger one, since that would fix the problem if it happens to have other causes in the future too.

**Comment by rsms** [comment]
It's a simple matter of hooking up a logger to the window's blur and focus event and then just verifying all the states and conditions.

**Comment by davepagurek** [comment]
Updated to verify the states. It's in a `setTimeout` right now to give time for an event to potentially be fired since I couldn't find a better way to ensure the state check happens after events are processed.

**Comment by davepagurek** [comment]
Updated to use upstream changes

**Comment by davepagurek** [comment]
Anything else that needs to be done on this?

**Comment by rsms** [comment]
Nice work!

**Comment by rsms** [comment]
Shipped in version 0.1.12

**Pull request merged**